src/Microsoft.OData.Core/Json/JsonWriter.cs

```diff
@@ -408,6 +409,137 @@ public void WriteValue(byte[] value)
             JsonValueUtils.WriteValue(this.writer, value, this.wrappedBuffer, this.ArrayPool);
         }
 
+#if NETCOREAPP3_1_OR_GREATER
+        public void WriteValue(JsonElement value)
+        {
+            switch (value.ValueKind)
+            {
+                case JsonValueKind.String:

```

Comment:
Would a Guid be handled as a string?

Reply:
Yes. Actually, JSON has no native support for Guids and dates, so those are all represented as strings. So, in such cases, we'll just write those back as strings. It will be up to the client on the other side, armed with a copy of the model, to figure out the correct type.